extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisAPIProducer.java

```diff
@@ -133,4 +107,24 @@ private String buildConnectionString(InetSocketAddress address) {
 
         return builder.toString();
     }
+
+    static class MutinyRedisClient extends io.vertx.mutiny.redis.client.Redis {
+        MutinyRedisClient(Redis delegate) {
+            super(delegate);
+        }
+
+        MutinyRedisClient() {
+            super(null);
+        }
+    }
+
+    static class MutinyRedisClientAPI extends io.vertx.mutiny.redis.client.RedisAPI {
+        MutinyRedisClientAPI(RedisAPI delegate) {
+            super(delegate);
+        }
+
+        MutinyRedisClientAPI() {
+            super(null);
+        }
+    }
```

Comment:
@gsmet I kind of need this to fix https://github.com/quarkusio/quarkus/pull/11354#issuecomment-673325574 as registering the classes for reflection did not solve the issue. 

I could not dig deeper what caused the issues though as I was out of depth lacking CDI and generated bytecode expertise ;-)

Let me know what you think of this.

Reply:
I have absolutely no idea :). @mkouba ?

Reply:
Hm, the no-args constructor defined on `io.vertx.mutiny.redis.client.Redis` is package-private so it cannot be accessed from a client proxy generated for a producer method declared on a class with different package... but I can see that you've tried to wrap the mutiny redis client in a special class. @machi1990 Do you still get the same `IllegalAccessError`?

Reply:
> Hm, the no-args constructor defined on `io.vertx.mutiny.redis.client.Redis` is package-private so it cannot be accessed from a client proxy generated for a producer method declared on a class with different package... 

I am wondering why a no arg ctor is needed in this case since the extension takes care of producing the instances by calling appropriate ctor. 

> but I can see that you've tried to wrap the mutiny redis client in a special class. @machi1990 Do you still get the same `IllegalAccessError`?

Wrapping it in a special class get's me past the error.

Reply:
> I am wondering why a no arg ctor is needed in this case since the extension takes care of producing the instances by calling appropriate ctor.

It's necessary to create a client proxy instance (used for normal scopes).

Reply:
Thanks for clarifying. 

You think the fix I used is an appropriate one?

Reply:
It's probably OK.

Reply:
Thanks. I suppose this PR is good to go then :-)